main.py

```diff
@@ -10,24 +11,47 @@
 
 # TODO probably needs to be moved to other file eventually
 HOST = args.host
-if( HOST == None):
+if(HOST == None):
     s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
     s.connect(('8.8.8.8', 1))  # connect() for UDP doesn't send 
     HOST = s.getsockname()[0]
 
 PORT = args.port # TODO validate port
-BUFFERSIZE = 1024
+BUFFERSIZE = 160
 s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
 s.bind((HOST,PORT))
 
 logging.info("Listening on %s:%d", HOST, PORT)
 # TODO temp code that repeatedly logs data from any connecting host
 while 1:
```

Comment:
surround this in a try? in case an error occurs it will ensure the connection is always closed:

```
try:
 while 1:
     // do some stuff
except KeyboardInterrupt:
    connection.close()
```

You will have to declare the connection above.